python/lsst/jointcal/jointcal.py

```diff
@@ -418,11 +445,16 @@ def _fit_photometry(self, associations):
             model : lsst.jointcal.PhotometryModel
                 The photometric model that was fit.
         """
-
         self.log.info("=== Starting photometric fitting...")
-        model = lsst.jointcal.SimplePhotometryModel(associations.getCcdImageList())
+
+        if self.config.photometryModel == "constrained":
+            model = lsst.jointcal.ConstrainedPhotometryModel(associations.getCcdImageList())
+        elif self.config.photometryModel == "simple":
+            model = lsst.jointcal.SimplePhotometryModel(associations.getCcdImageList())
```

Comment:
Feel free to defer this to another ticket, but I smell a use for `pex.config.RegistryField` here.